feelpp/feel/feeldiscr/functionspace.hpp

```diff
@@ -5482,6 +5487,182 @@ public std::enable_shared_from_this<FunctionSpace<A0,A1,A2,A3,A4> >
+
+            // fetch current mapping : (pt ids in elt) -> (dof in process ordered by local dof)
+            using doftable_relation_type = std::unordered_map<std::vector /*set*/<index_type>, std::vector<index_type>, Feel::HashTables::HasherContainers<index_type>>;
+            doftable_relation_type currentDofTableMapping;
+            std::vector<index_type> ptIds;
+            std::vector<index_type> dd;
+            auto meshsupport = this->template meshSupport<0>();
+            auto dof = this->dof();
+            for ( auto const& eltWrap : elements(meshsupport) )
+            {
+                auto const& elt = unwrap_ref(eltWrap);
+                ptIds.resize( elt.nVertices() );
+                for ( int p=0;p<elt.nVertices();++p )
+                    ptIds[p] = elt.point(p).id();
+
+                auto dofMapping = dof->localDof( elt.id() );
+                dd.resize( std::distance( Feel::begin(dofMapping),Feel::end(dofMapping ) ) );
+                for( auto const& ldof : dofMapping )
+                {
+                    index_type thedof = ldof.second.index();
+                    uint16_type thelocdof = ldof.first.localDof();
+                    dd[thelocdof] = thedof;
+                }
+
+                currentDofTableMapping.insert( {ptIds, dd} );
+            }
+
+            // read hdf5 file
+            HDF5 hdf5;
+#if 0
+            hdf5.openFile( h5filepath.string(), (subComm)? *subComm : this->comm().comm(), true );
+#else
+            hdf5.openFile( h5filepath.string(), this->worldComm().comm(), true );
+#endif
+            std::string tableName = "doftable";
+            hsize_t dimsGlob[2];
+            hsize_t offsetElt[2] = {0,0};
+            hdf5.openTable( tableName, dimsGlob );
+
+            std::vector<uint> dataReaded( dimsGlob[0]*dimsGlob[1] );
+
+            hdf5.read( tableName, H5T_NATIVE_UINT, dimsGlob, offsetElt, dataReaded.data() );
+
+            hdf5.closeTable( tableName );
+            hdf5.closeFile();
+
+            // update mapping
+            uint16_type nVerticesInElt = ptIds.size();
+            uint16_type nDofByElt = dd.size();
+            std::vector<index_type> mappingWithFile( dof->nLocalDofWithGhost(), invalid_v<index_type> );
+
+            for ( size_type k=0; k<dataReaded.size(); )
+            {
+                index_type eltId = dataReaded[k++];
+                ptIds.resize( nVerticesInElt );
+                for ( int p=0;p<nVerticesInElt;++p )
+                    ptIds[p] = dataReaded[k++];
+                auto const& curentLpDofs = currentDofTableMapping.at( ptIds );
+
+                for ( int ld=0;ld<nDofByElt;++ld )
+                    mappingWithFile[ curentLpDofs[ld] ] = dataReaded[k++];
+            }
+
+            return mappingWithFile;
+        }
+
+    template <typename TT=functionspace_type,std::enable_if_t< TT::is_composite, bool> = true >
+    std::vector<index_type> relationFromFile( std::string const& filepathstr ) const
```

Comment:
why not use if constexpr instead of SFINAE ? the code would be a tad more simpler/readable